src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/OrderingHelper.cs

```diff
@@ -0,0 +1,361 @@
+        /// </summary>
+        /// <param name="projectTree">the given project tree</param>
+        /// <param name="returnSibling">passes the index of the given project tree from the given ordered sequence, expecting to return a sibling</param>
+        /// <returns>a sibling</returns>
+        private static IProjectTree2 GetSiblingByDisplayOrder(IProjectTree projectTree, Func<int, IEnumerable<IProjectTree>, IProjectTree2> returnSibling)
+        {
+            var parent = projectTree.Parent;
+            var displayOrder = GetDisplayOrder(projectTree);
+            if (!IsValidDisplayOrder(displayOrder) || parent == null)
+            {
+                return null;
+            }
+
+            var orderedChildren = parent.Children.Where(x => IsValidDisplayOrder(GetDisplayOrder(x))).OrderBy(x => GetDisplayOrder(x)).ToArray();
+
+            for (var i = 0; i < orderedChildren.Length; ++i)
+            {
+                var sibling = orderedChildren[i];
+                if (GetDisplayOrder(sibling) == displayOrder)
+                {
+                    return returnSibling(i, orderedChildren);
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets the previous sibling of the given project tree, if there is any. Can return null.
+        /// </summary>
+        private static IProjectTree2 GetPreviousSibling(IProjectTree projectTree)
+        {
+            return GetSiblingByDisplayOrder(projectTree, (i, orderedChildren) =>
+            {
+                if (i == 0)
+                {
+                    return null;
+                }
+
+                return orderedChildren.ElementAt(i - 1) as IProjectTree2;
+            });
+        }
+
+        /// <summary>
+        /// Gets the next sibling of the given project tree, if there is any. Can return null.
+        /// </summary>
+        private static IProjectTree2 GetNextSibling(IProjectTree projectTree)
+        {
+            return GetSiblingByDisplayOrder(projectTree, (i, orderedChildren) =>
+            {
+                if (i == (orderedChildren.Count() - 1))
+                {
+                    return null;
+                }
+
+                return orderedChildren.ElementAt(i + 1) as IProjectTree2;
+            });
+        }
+
+        /// <summary>
+        /// Gets a sibling of the given project tree based on the move action. Can return null.
+        /// </summary>
+        private static IProjectTree GetSiblingByMoveAction(IProjectTree projectTree, MoveAction moveAction)
+        {
+            switch (moveAction)
+            {
+                case MoveAction.Up:
+                    return GetPreviousSibling(projectTree);
+
+                case MoveAction.Down:
+                    return GetNextSibling(projectTree);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets a reference element based on the given project tree and move action. Can return null.
```

Comment:
I have no idea what a "reference element" means.

Reply:
I described what it was in TryMove. I'll move the description here.